cmd/aws/create_test.go

```diff
@@ -0,0 +1,417 @@
+	if m.fnRetrieve == nil {
+		return aws.Credentials{}, errors.New("not implemented")
+	}
+
+	creds, err := m.fnRetrieve(ctx)
+	if err != nil {
+		return aws.Credentials{}, err
+	}
+	return *creds, nil
+}
+
+type mockInstanceTypesPaginator struct {
+	instanceTypes []ec2Types.InstanceTypeInfo
+	err           error
+	called        bool
+}
+
+func (m *mockInstanceTypesPaginator) HasMorePages() bool {
+	return !m.called
+}
+
+func (m *mockInstanceTypesPaginator) NextPage(ctx context.Context, opts ...func(*ec2.Options)) (*ec2.DescribeInstanceTypesOutput, error) {
+	if m.err != nil {
+		return nil, m.err
+	}
+	m.called = true
+	return &ec2.DescribeInstanceTypesOutput{
+		InstanceTypes: m.instanceTypes,
+	}, nil
+}
+
+type mockSSMClient struct {
+	ssm.Client
+	parameterValue string
+	err            error
+}
+
+func (m *mockSSMClient) GetParameter(ctx context.Context, input *ssm.GetParameterInput, opts ...func(*ssm.Options)) (*ssm.GetParameterOutput, error) {
+	if m.err != nil {
+		return nil, m.err
+	}
+	return &ssm.GetParameterOutput{
+		Parameter: &ssmTypes.Parameter{
+			Value: &m.parameterValue,
+		},
+	}, nil
+}
+
+type mockEC2Client struct {
+	ec2.Client
+	images       []ec2Types.Image
+	architecture string
+	err          error
+}
+
+func (m *mockEC2Client) DescribeImages(ctx context.Context, input *ec2.DescribeImagesInput, opts ...func(*ec2.Options)) (*ec2.DescribeImagesOutput, error) {
+	if m.err != nil {
+		return nil, m.err
+	}
+	if m.architecture == "x86_64" {
+		return &ec2.DescribeImagesOutput{
+			Images: []ec2Types.Image{
+				{
+					Architecture: ec2Types.ArchitectureValues("x86_64"),
+				},
+			},
+		}, nil
+	} else if m.architecture == "" {
+		return &ec2.DescribeImagesOutput{
+			Images: m.images,
+		}, nil
+	} else {
+		return nil, errors.New("unexpected architecture")
+	}
+}
+
+func TestValidateCredentials(t *testing.T) {
+	tests := []struct {
+		name        string
+		creds       aws.Credentials
+		err         error
+		expectedErr error
+	}{
+		{
+			name: "valid credentials",
+			creds: aws.Credentials{
+				AccessKeyID:     "test-access-key-id",
+				SecretAccessKey: "test-secret-access-key",
+				SessionToken:    "test-session-token",
+			},
+			err:         nil,
+			expectedErr: nil,
+		},
+		{
+			name:        "failed to retrieve credentials",
+			creds:       aws.Credentials{},
+			err:         errors.New("failed to retrieve credentials"),
+			expectedErr: errors.New("failed to retrieve AWS credentials: failed to retrieve credentials"),
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			mockProvider := &mockCredentialsProvider{
+				fnRetrieve: func(ctx context.Context) (*aws.Credentials, error) {
+					return &tt.creds, tt.err
+				},
+			}
+
+			creds, err := getSessionCredentials(context.Background(), mockProvider)
+			if tt.expectedErr != nil {
+				require.Nil(t, creds)
+				require.EqualError(t, err, tt.expectedErr.Error())
```

Comment:
I mentioned in the previous review but it seems it was marked as resolved without actually resolving it. This is a very weak validation that depends on the fact that a) both haveto be error types, and b) both error string versions have to match. Not sure you want to do that for the reasons I provided before.

The same applies for the rest of the checks in the unit tests below.